Refresh.GameServer/Endpoints/Game/Playlists/PlaylistEndpoints.cs

```diff
@@ -0,0 +1,338 @@
+                dataContext.Database.GetPlaylistsByAuthorContainingPlaylist(author, playlist).ToList();
+        }
+        else
+        {
+            GameLevel? level = dataContext.Database.GetLevelByIdAndType(slotType, slotId);
+
+            if (level == null)
+                return null;
+            
+            playlists = author == null ? 
+                dataContext.Database.GetPlaylistsContainingLevel(level).ToList() : 
+                dataContext.Database.GetPlaylistsByAuthorContainingLevel(author, level).ToList();
+        }
+        
+        int total = playlists.Count;
+        
+        (int skip, int count) = context.GetPageData();
+
+        // Return the serialized playlists 
+        return new SerializedMinimalLevelList(
+            GameMinimalLevelResponse.FromOldList(playlists.Skip(skip).Take(count), dataContext), 
+            total, 
+            skip
+        );
+    }
+
+    [GameEndpoint("setPlaylistMetaData/{id}", HttpMethods.Post, ContentType.Xml)]
+    [RequireEmailVerified]
+    public Response UpdatePlaylistMetadata(RequestContext context, GameDatabaseContext database, GameUser user, int id, SerializedPlaylist body)
+    {
+        GamePlaylist? playlist = database.GetPlaylistById(id);
+
+        // Handle a bad playlist ID
+        if (playlist == null)
+            return NotFound;
+
+        // Dont allow the wrong user to update playlists
+        if (playlist.Creator.UserId != user.UserId)
+            return Unauthorized;
+        
+        database.UpdatePlaylist(playlist, body);
+        
+        return OK;
+    }
+
+    [GameEndpoint("deletePlaylist/{id}", HttpMethods.Post)]
+    [RequireEmailVerified]
+    public Response DeletePlaylist(RequestContext context, GameDatabaseContext database, GameUser user, int id)
+    {
+        GamePlaylist? playlist = database.GetPlaylistById(id);
+
+        // Handle a bad playlist ID
+        if (playlist == null)
+            return NotFound;
+
+        // Dont allow the wrong user to delete playlists
+        if (playlist.Creator.UserId != user.UserId)
+            return Unauthorized;
+
+        database.DeletePlaylist(playlist);
+            
+        return OK;
+    }
+
+    [GameEndpoint("addToPlaylist/{playlistId}", HttpMethods.Post)]
+    [RequireEmailVerified]
+    public Response AddSlotToPlaylist(RequestContext context, GameDatabaseContext database, GameUser user, int playlistId)
+    {
+        // Extract the slot type, ensuring its set
+        string? slotType = context.QueryString["slot_type"];
+        if (slotType == null)
+            return BadRequest;
+        
+        // Extract the slot ID ensuring its valid
+        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
+            return BadRequest;
+
+        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
+
+        // If the parent doesn't exist, exit gracefully
+        if (parentPlaylist == null)
+            return NotFound;
```

Comment:
Alright, we need to have a talk about these comments. They should be used for when there's weird non-standard behavior going on or something, or something actually needs to be described that isn't obvious from just reading the code. We don't need a comment on every single branch like this.

Everyone can see that if `TryParse` returns false then it exits, for example.

```suggestion
        string? slotType = context.QueryString["slot_type"];
        if (slotType == null)
            return BadRequest;
        
        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
            return BadRequest;

        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
        if (parentPlaylist == null)
            return NotFound;
```

Reply:
Main reason why i did it like that here is that i wrote the comments before the code to help me visualize fully what i need to do before i put the time into actually adding the missing functions/functionality needed to implement it, and to be a bit more mentally careful of error handling. After writing the code around the comments, i just kinda forgot about them and left them mostly in tact, i'll be sure to trim them down next time i do it that way.